src/components/taskDetails/Details.tsx

```diff
@@ -1,20 +1,86 @@
-import React, { FC, useState } from 'react';
+import React, { FC, useEffect, useState } from 'react';
 import moment from 'moment';
 import { FaReceipt } from 'react-icons/fa6';
 import Link from 'next/link';
 import Tooltip from '@/components/common/Tooltip/Tooltip';
 import setColor from './taskPriorityColors';
 import extractRepoName from '@/utils/extractRepoName';
 import styles from './task-details.module.scss';
-import { TaskDetailsProps } from '@/interfaces/taskDetails.type';
+import {
+    DetailsContentProps,
+    TaskDetailsProps,
+} from '@/interfaces/taskDetails.type';
+import useUserData from '@/hooks/useUserData';
 import { STARTED_ON, ENDS_ON } from '@/constants/constants';
 
 type StringNumberOrUndefined = string | number | undefined;
 
-const Details: FC<TaskDetailsProps> = ({ detailType, value, url }) => {
+const DetailsContent: FC<DetailsContentProps> = ({
+    color,
+    isGitHubLink,
+    value,
+    gitHubIssueLink,
+    isTimeDetail,
+    formatDate,
+    tooltipActive,
+    renderedValue,
+    getRelativeTime,
+}) => {
+    return (
+        <span
+            className={styles.detailValue}
+            style={{ color: color ?? 'black' }}
+        >
+            {isGitHubLink && value ? (
+                <a
+                    className={styles.gitLink}
+                    href={gitHubIssueLink}
+                    target="_blank"
+                    rel="noopener noreferrer"
+                    aria-label="Open GitHub Issue"
+                    title={value}
+                >
+                    {isGitHubLink ? `${extractRepoName(value)}` : value}
+                </a>
+            ) : isTimeDetail ? (
+                <Tooltip
+                    content={formatDate(value)}
+                    tooltipPosition={{
+                        top: '-3.6rem',
+                        right: '-4.5rem',
+                    }}
+                >
+                    {tooltipActive ? formatDate(value) : getRelativeTime(value)}
+                </Tooltip>
+            ) : (
+                renderedValue
+            )}
+        </span>
+    );
+};
+
+const Details: FC<TaskDetailsProps> = (props) => {
+    const { detailType, value, url, isEditing, setEditedTaskDetails } = props;
     const color = value ? setColor?.[value] : undefined;
     const isGitHubLink = detailType === 'Link';
     const gitHubIssueLink = isGitHubLink ? value : undefined;
+    const [newEndOnDate, setNewEndOnDate] = useState('');
+    const { isUserAuthorized } = useUserData();
+
+    useEffect(() => {
+        if (!isEditing && newEndOnDate !== '') setNewEndOnDate('');
+    }, [isEditing, newEndOnDate]);
```

Comment:
`useEffect` is triggered whenever either `isEditing` or `newEndOnDate` changes, but logically, you only need to reset the date when `isEditing` changes from `true` to `false`.


You can simplify this by using `useEffect` only for `isEditing`:
```suggestion
   useEffect(() => {
    if (!isEditing) setNewEndOnDate('');
}, [isEditing]);

```


This will avoid unnecessary updates and improve performance slightly by preventing the `setNewEndOnDate` function from being called on every render when the date changes.

Reply:
This is what I did previously

Reply:
https://github.com/Real-Dev-Squad/website-status/pull/1260#discussion_r1769262587

Reply:
reverted and pushed changes this way